sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -252,29 +242,59 @@ public void remove(String propertyName) {
      * @param value the value of the property.
      */
     @SuppressWarnings({"unchecked", "rawtypes"})
-    public <T> void set(String propertyName, T value) {
+    public <T> void set(String propertyName, T value, CosmosItemSerializer itemSerializer) {
+        checkNotNull(itemSerializer, "Argument 'itemSerializer' must not be null.");
+        checkNotNull(itemSerializer, "Argument 'itemSerializer' must be the DEFAULT_SERIALIZER when using this method.");
```

Comment:
Shouldn't `checkNotNull` in line 247 be a check for type which is `DefaultCosmosItemSerializer`? Also, if line 246 fails, we would exit early.